dynamiqs/mesolve/mesolve.py

```diff
@@ -182,25 +189,18 @@ def _check_mesolve_args(
     tsave: Array,
     exp_ops: Array | None,
 ):
-    # === check H shape
     check_shape(H, 'H', '(?, n, n)', subs={'?': 'nH?'})
+    check_shape(rho0, 'rho0', '(?, n, 1)', '(?, n, n)', subs={'?': 'nrho0?'})
+    check_times(tsave, 'tsave')
+
+    if exp_ops is not None:
+        check_shape(exp_ops, 'exp_ops', '(N, n, n)', subs={'N': 'nE'})
+
+    for i, L in enumerate(jump_ops):
+        check_shape(L, f'jump_ops[{i}]', '(?, n, n)', subs={'?': 'nL?'})
```

Comment:
Nit: it'd be better to order this function as before, by arguments order as in the function signature (for code readability + if the user made multiple mistakes, he'll see them in order and fix the function call left to right).